src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsSteps.cs

```diff
@@ -0,0 +1,167 @@
+            this.posReceiverPrivateKey = wallet.GetExtendedPrivateKeyForAddress(PosWalletPassword, this.posReceiverAddress).PrivateKey;
+
+            this.nodes[PosStaker].SetDummyMinerSecret(new BitcoinSecret(this.posReceiverPrivateKey, this.nodes[PosStaker].FullNode.Network));
+            this.nodeGroupBuilder.WithConnections().Connect(PowMiner, PosStaker);
+        }
+
+        private void sends_a_million_coins_from_pow_wallet_to_pos_wallet()
+        {
+            var context = SharedSteps.CreateTransactionBuildContext(
+                PowWallet,
+                WalletAccount,
+                PowWalletPassword,
+                this.posReceiverAddress.ScriptPubKey,
+                Money.COIN * 1000000,
+                FeeType.Medium,
+                101);
+
+            context.OverrideFeeRate = new FeeRate(Money.Satoshis(20000));
+
+            var unspent = this.nodes[PowMiner].FullNode.WalletManager().GetSpendableTransactionsInWallet(PowWallet);
+            var coins = new List<Coin>();
+
+            var blockTimestamp = unspent.OrderBy(u => u.Transaction.CreationTime).Select(ts => ts.Transaction.CreationTime).First();
+            var transaction = new Transaction
+            {
+                Time = (uint)blockTimestamp.ToUnixTimeSeconds()
+            };
+
+            foreach (var item in unspent.OrderByDescending(a => a.Transaction.Amount))
+            {
+                coins.Add(new Coin(item.Transaction.Id, (uint)item.Transaction.Index, item.Transaction.Amount, item.Transaction.ScriptPubKey));
+            }
+
+            var coin = coins.First();
+            var txIn = transaction.AddInput(new TxIn(coin.Outpoint, this.powSenderAddress.ScriptPubKey));
+            transaction.AddOutput(new TxOut(new Money(9699999999995400), this.powSenderAddress.ScriptPubKey));
+            transaction.AddOutput(new TxOut(new Money(100000000000000), this.posReceiverAddress.ScriptPubKey));
+            transaction.Sign(this.powSenderPrivateKey, new[] { coin });
+
+            this.nodes[PowMiner].FullNode.NodeService<WalletController>().SendTransaction(new SendTransactionRequest(transaction.ToHex()));
+        }
+
+        private void pow_wallet_broadcasts_tx_of_million_coins_and_pos_wallet_receives()
+        {
+            TestHelper.WaitLoop(() => this.nodes[PosStaker].CreateRPCClient().GetRawMempool().Length > 0);
+            TestHelper.WaitLoop(() => this.nodes[PosStaker].FullNode.WalletManager().GetSpendableTransactionsInWallet(PosWallet).Any());
+
+            this.sharedSteps.WaitForNodeToSync(this.nodes[PosStaker]);
+
+            var received = this.nodes[PosStaker].FullNode.WalletManager().GetSpendableTransactionsInWallet(PosWallet);
+            received.Sum(s => s.Transaction.Amount).Should().Be(Money.COIN * 1000000);
+        }
+
+        private void pos_node_mines_ten_blocks_more_ensuring_they_can_be_staked()
+        {
+            this.nodes[PosStaker].GenerateStratisWithMiner(10);
```

Comment:
Ideally put the ConsensusOptions.Maturity (instead of had coding 10)

Reply:
Fixed!